src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java

```diff
@@ -15,191 +11,232 @@
+            if (searchUserResult != null && Constants.OK.equalsIgnoreCase(searchUserResult.getResponseCode())
+                    && searchUserResult.getResult().getResponse().getCount() > 0) {
+                for (SearchUserApiContent searchUserApiContent : searchUserResult.getResult().getResponse()
+                        .getContent()) {
+                    result.put(searchUserApiContent.getUserId(), searchUserApiContent);
+                }
+            }
+        } catch (Exception e) {
+            throw new ApplicationLogicError("Sunbird Service ERROR: ", e);
+        }
+
+        return result;
+    }
+
+    @Override
+    public Map<String, Object> updateLogin(LastLoginInfo userLoginInfo) {
+        Map<String, Object> response = new HashMap<>();
+        logger.info(String.format("Updating User Login: rootOrg: %s, userId: %s", userLoginInfo.getOrgId(),
+                userLoginInfo.getUserId()));
+        userLoginInfo.setLoginTime(new Timestamp(new Date().getTime()));
+        try {
+            Map<String, Object> propertyMap = new HashMap<>();
+            propertyMap.put(Constants.USER_ID, userLoginInfo.getUserId());
+            List<Map<String, Object>> details = cassandraOperation.getRecordsByProperties(Constants.DATABASE,
+                    Constants.LOGIN_TABLE, propertyMap, null);
+            if (CollectionUtils.isEmpty(details)) {
+                Map<String, Object> request = new HashMap<>();
+                request.put(Constants.USER_ID, userLoginInfo.getUserId());
+                request.put(Constants.FIRST_LOGIN_TIME, userLoginInfo.getLoginTime());
+                cassandraOperation.insertRecord(Constants.DATABASE, Constants.LOGIN_TABLE, request);
+                telemetryUtils.pushDataToKafka(telemetryUtils.getUserslastLoginTelemetryEventData(userLoginInfo),
+                        serverConfig.getUserUtilityTopic());
+            }
+        } catch (Exception e) {
+            throw new ApplicationLogicError("Sunbird Service ERROR: ", e);
+        }
+        response.put(userLoginInfo.getUserId(), Boolean.TRUE);
+        return response;
+    }
+
+    @Override
+    public Map<String, Object> getUsersReadData(String userId, String authToken, String X_authToken) {
+        Map<String, String> header = new HashMap<>();
+        header.put(Constants.AUTH_TOKEN, authToken);
+        header.put(Constants.X_AUTH_TOKEN, X_authToken);
+        Map<String, Object> readData = (Map<String, Object>) outboundRequestHandlerService
+                .fetchUsingGetWithHeadersProfile(serverConfig.getSbUrl() + serverConfig.getLmsUserReadPath() + userId,
+                        header);
+        Map<String, Object> result = (Map<String, Object>) readData.get(Constants.RESULT);
+        Map<String, Object> responseMap = (Map<String, Object>) result.get(Constants.RESPONSE);
+        return responseMap;
+    }
+
+    @Override
+    public Map<String, Object> autoSearchUser(String key) throws IOException {
```

Comment:
Need overridden method which includes requestBody -- "query" parameter in the request body should be used as "key".